.github/workflows/deploy.yml

```diff
@@ -0,0 +1,99 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+#
+# See documentation here: https://github.com/aws-actions/amazon-ecs-deploy-task-definition
+
+# GitHub recommends pinning actions to a commit SHA.
+# To get a newer version, you will need to update the SHA.
+# You can also reference a tag or branch, but the action may change without warning.
+
+name: Deploy to Amazon ECS
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  # These values come from https://github.com/anthroarts/fargate-infrastructure
+  AWS_REGION: us-west-2
+  AWS_ROLE: arn:aws:iam::474196809376:role/engelsystem-github-deploy
+  CONTAINER_NAME: engelsystem
+  ECR_REPOSITORY: engelsystem
+  ECS_CLUSTER: Fargate
+  ECS_SERVICE: engelsystem
+  TASK_DEFINITION_FAMILY: engelsystem
+  # These should be updated as you develop
+  MAJOR_VERSION: 1
+  MINOR_VERSION: 0
+  PATCH_VERSION: ${{ github.run_number }}
+  # Set this to true only if you want to override engelsystem terraform fargate tag configuration.
+  # IE, you want to force a deployment and ignore whatever is specified for the tag inside
+  # of https://github.com/anthroarts/fargate-infrastructure/blob/main/engelsystem_ecs.tf
+  WRITE_NEW_TASK_DEFINITION_TAG: false
+
+permissions:
+  # This is required for requesting the OIDC credentials
+  # https://github.com/aws-actions/configure-aws-credentials?tab=readme-ov-file#oidc
+  id-token: write
+  contents: read
+
+jobs:
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ env.AWS_ROLE }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build, tag, and push image to Amazon ECR
+        id: build-image
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          IMAGE_TAG: ${{ github.sha }}
+        run: |
+          # Build a docker container and
+          # push it to ECR so that it can
+          # be deployed to ECS.
+          docker build -f docker/Dockerfile . \
+            -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG \
+            -t $ECR_REGISTRY/$ECR_REPOSITORY:$MAJOR_VERSION \
+            -t $ECR_REGISTRY/$ECR_REPOSITORY:$MAJOR_VERSION.$MINOR_VERSION \
+            -t $ECR_REGISTRY/$ECR_REPOSITORY:$MAJOR_VERSION.$MINOR_VERSION.$PATCH_VERSION \
+            -t $ECR_REGISTRY/$ECR_REPOSITORY:latest
```

Comment:
NB: I thought `ECR_REPOSITORY` is the name of the repo/registry. I didn't pay enough attention earlier. I think it would be more intuitive to rename this to `ENGELSYSTEM_DOCKER_IMAGE_NAME` or just `DOCKER_IMAGE_NAME` - what do you think?

Reply:
The ECR_REPOSITORY and ECR_REGISTRY come from https://docs.github.com/en/actions/use-cases-and-examples/deploying/deploying-to-amazon-elastic-container-service#creating-the-workflow

I don't want to go around renaming things too much if people are referencing those official docs.